parse.go

```diff
@@ -49,12 +50,29 @@ func NewParser(in io.Reader, bytesToRead int64, frameChannel chan *frame.Frame)
 		decoder:      buffer,
 		frameChannel: frameChannel,
 	}
+	metaElems := p.parseFileHeader()
+	if p.decoder.Error() != nil {
+		return nil, p.decoder.Error()
+	}
+	p.parsedElements = &element.DataSet{Elements: metaElems}
 
+	return &p, nil
+}
+
+// NewParser initializes and returns a new Parser with the 'DICM' keyword check overriden 
+func NewParserForce(in io.Reader, bytesToRead int64, frameChannel chan *frame.Frame) (Parser, error) {
```

Comment:
I wasn't quite sure what to name it, but I like `NewParserWithForcedRead` better too. I'll change it

Reply:
@IChocked @suyashkumar I provided this feedback already, and I don't know how streaming DICOMs is popular, but my $0.02: `dicom.NewParserFromBytesWithForcedRead` should be the 1 main function. Then `dicom.NewParserFromFileWithForcedRead` should use it. Then `dicom.NewParserFromBytes` should just call `dicom.NewParserFromBytesWithForcedRead` with `force=0`, and same for `dicom.NewParserFromFile`. In other words: since we often read files from memory, this fix won't really work for us, unless I'm missing how `NewParserFromBytes` can get this "force=1" flag.

I think it's what Suyash is mentioning below. I'm just saying here explicitly that I care about "FromBytes" very much :-)